backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -416,20 +416,24 @@ boolean childContains(Set<?> list, Set<?> sublist) {
       deletedLocations.removeAll(locationBlobsMapCommitB.keySet());
       LOGGER.debug("Deleted location for Diff : {}", deletedLocations);
 
-      // modified new blob location from the CommitA
-      Set<String> modifiedLocations = new LinkedHashSet<>(locationBlobsMapCommitB.keySet());
-      modifiedLocations.removeAll(addedLocations);
-      Map<String, BlobExpanded> commonBlobs =
-          locationBlobsMapCommitA.values().stream().collect(toMap(Entry::getValue, Entry::getKey));
-      commonBlobs
-          .keySet()
-          .retainAll(
-              locationBlobsMapCommitB.values().stream()
-                  .map(Entry::getValue)
-                  .collect(Collectors.toSet()));
-      Map<String, BlobExpanded> locationBlobsCommon =
-          getLocationWiseBlobExpandedMapFromCollection(commonBlobs.values());
-      modifiedLocations.removeAll(locationBlobsCommon.keySet());
+      // get B sha -> blobs
+      Map<String, Set<BlobExpanded>> blobsB = getCollectToMap(locationBlobsMapCommitB);
+      // get A sha -> blobs
+      Map<String, Set<BlobExpanded>> blobsA = getCollectToMap(locationBlobsMapCommitA);
+      // delete blobs same with A
+      for (Map.Entry<String, Set<BlobExpanded>> entry : blobsA.entrySet()) {
+        Set<BlobExpanded> ent = blobsB.get(entry.getKey());
+        if (ent != null) {
+          ent.removeAll(entry.getValue());
```

Comment:
please make sure that call by reference updates the value in `blobsB` map.

Reply:
it does